lib/ansible/template/vars.py

```diff
@@ -57,6 +57,22 @@ def __init__(self, templar, globals, locals=None, *extras):
                     elif key not in ('context', 'environment', 'template'):
                         self._locals[key] = val
 
+    def __iter__(self):
+        for k in self._templar._available_variables:
```

Comment:
The existing AnsibleJ2Vars code seems to go out of it's way to avoid copys (aside from it's ._locals) 
```
Added bonus is avoiding duplicating the large
    hashes that inject tends to be.
```
But first glance, the Mapping in https://github.com/zmedico/ansible/pull/1/files seems to be all references without copies so seems like it should be okay to me, but haven't tested.